cocotb/share/include/cocotb_bfm_api.h

```diff
@@ -0,0 +1,142 @@
+/******************************************************************************
+* Copyright (c) 2013 Potential Ventures Ltd
+* Copyright (c) 2013 SolarFlare Communications Inc
+* Copyright (c) 2019 Matthew Ballance
+* All rights reserved.
+*
+* Redistribution and use in source and binary forms, with or without
+* modification, are permitted provided that the following conditions are met:
+*    * Redistributions of source code must retain the above copyright
+*      notice, this list of conditions and the following disclaimer.
+*    * Redistributions in binary form must reproduce the above copyright
+*      notice, this list of conditions and the following disclaimer in the
+*      documentation and/or other materials provided with the distribution.
+*    * Neither the name of Potential Ventures Ltd,
+*       SolarFlare Communications Inc nor the
+*      names of its contributors may be used to endorse or promote products
+*      derived from this software without specific prior written permission.
+*
+* THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+* ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+* WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+* DISCLAIMED. IN NO EVENT SHALL POTENTIAL VENTURES LTD BE LIABLE FOR ANY
+* DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+* (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+* LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+* ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+* (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+* SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+******************************************************************************/
+#ifndef INCLUDED_COCOTB_BFM_API_H
+#define INCLUDED_COCOTB_BFM_API_H
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+// Callback function type used to notify
+// implementation when a message is available
+typedef void (*cocotb_bfm_notify_f)(void *);
+
+int cocotb_bfm_register(
+		const char				*type_name,
```

Comment:
Nit: I haven't seen tabs used elsewhere AFAIR.

Reply:
Good point. I've converted tabs to spaces in the C/C++ files